### PR TITLE
Enforce Go linting policy status sync

### PR DESCRIPTION
**Pull request by ChunxiAlexLuo**
Signed-off-by: Chunxi Luo chuluo@redhat.com

https://github.com/open-cluster-management/backlog/issues/18035

### CODE REVIEW

File: controllers/sync/eventMapper.go [hunk 1]
```diff
@@ -13,7 +13,11 @@ import (
 )
 
 func eventMapper(obj client.Object) []reconcile.Request {
-	event := obj.(*corev1.Event)
+	event, eventOK := obj.(*corev1.Event)
+	if !eventOK {
+		panic(eventOK)
+	}
```
Comment (JustinKuli) [inline controllers/sync/eventMapper.go:19]: I think `if !ok {panic}` is basically the current behavior if the assertion isn't checked and fails. If we're confident the assertion won't fail, I'd rather have a comment to ignore the lint on this line.

```
//nolint:forcetypeassert
```

Comment (JustinKuli) [reply controllers/sync/eventMapper.go:19]: Or, looking at other cases like this in the PR, should it return an empty slice back?

File: controllers/sync/policy_status_sync.go [hunk 2]
```diff
@@ -131,17 +138,15 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
...
 	}
 	// filter events to current policy instance and build map
 	eventForPolicyMap := make(map[string]*[]policiesv1.ComplianceHistory)
-	rgx, err := regexp.Compile(`(?i)^policy:\s*([A-Za-z0-9.-]+)\s*\/([A-Za-z0-9.-]+)`)
-	if err != nil {
-		// regexp is wrong, how?
-		return reconcile.Result{}, err
-	}
+	// panic if regexp invalid
+	rgx := regexp.MustCompile(`(?i)^policy:\s*([A-Za-z0-9.-]+)\s*\/([A-Za-z0-9.-]+)`)
```
Comment (JustinKuli) [inline controllers/sync/policy_status_sync.go:149]: I like this, if we mess up the regexp then it should panic during a test, which will be more obvious than it was before 👍

**Review by JustinKuli — APPROVED** [review]
/hold

Just one optional change. Feel free to un-hold if you'd rather keep the (probably unreachable) panic.

File: controllers/sync/eventMapper.go (hunk 1 above)
Comment (ChunxiAlexLuo) [reply controllers/sync/eventMapper.go:19]: @JustinKuli I think the second suggestion may be a better idea, I will polish the log here in another logger pr

Comment (ChunxiAlexLuo) [reply controllers/sync/eventMapper.go]: @JustinKuli It seems an empty slice will false fail test cases, not look good to me on the corner case handling. I ignored this lint here and will take a further looking if need to open a new pr for it.

Comment (JustinKuli) [reply controllers/sync/eventMapper.go]: Awesome, thanks for checking!

**Review by JustinKuli — APPROVED** [review]
